[FX] Emit named tuple construction node when NamedTuple appears as an arg

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49553 [FX] Emit named tuple construction node when NamedTuple appears as an arg**

Previously, when a NamedTuple value was used in args/kwargs and the NamedTuple definition was in a different module (necessitating an import statement), we would miss 1) Adding the import statement and 2) having the repr() that refers to the right fully-qualified name, since the default repr for NamedTuple in Python does not print the FQN. This patch makes it so that `create_arg` emits a `call_function` node to the NamedTuple constructor and uses that `Node` as the arg, rather than embedding the NamedTuple arg directly



Differential Revision: [D25618577](https://our.internmc.facebook.com/intern/diff/D25618577)

Commit:
[FX] Emit named tuple construction node when NamedTuple appears as an arg

[ghstack-poisoned]

Commit:
Update on "[FX] Emit named tuple construction node when NamedTuple appears as an arg"


Differential Revision: [D25618577](https://our.internmc.facebook.com/intern/diff/D25618577)

[ghstack-poisoned]

zdevito reviewed (approved):
Cool this is very simple. There are 3 lines in map_arg for named tuples that can be deleted as well.

Commit:
Update on "[FX] Emit named tuple construction node when NamedTuple appears as an arg"


Previously, when a NamedTuple value was used in args/kwargs and the NamedTuple definition was in a different module (necessitating an import statement), we would miss 1) Adding the import statement and 2) having the repr() that refers to the right fully-qualified name, since the default repr for NamedTuple in Python does not print the FQN. This patch makes it so that `create_arg` emits a `call_function` node to the NamedTuple constructor and uses that `Node` as the arg, rather than embedding the NamedTuple arg directly



Differential Revision: [D25618577](https://our.internmc.facebook.com/intern/diff/D25618577)

[ghstack-poisoned]
